RecoBTag/ONNXRuntime/plugins/BoostedJetONNXJetTagsProducer.cc

```diff
@@ -63,75 +54,18 @@ class BoostedJetONNXJetTagsProducer : public edm::stream::EDProducer<edm::Global
-        prep_params.min_length = group_pset.at("min_length");
-        prep_params.max_length = group_pset.at("max_length");
-        input_shapes_.push_back({1, (int64_t)prep_params.var_names.size(), -1});
-      }
-      const auto &var_info_pset = group_pset.at("var_infos");
-      for (const auto &var_name : prep_params.var_names) {
-        const auto &var_pset = var_info_pset.at(var_name);
-        double median = var_pset.at("median");
-        double norm_factor = var_pset.at("norm_factor");
-        double replace_inf_value = var_pset.at("replace_inf_value");
-        double lower_bound = var_pset.at("lower_bound");
-        double upper_bound = var_pset.at("upper_bound");
-        double pad = var_pset.contains("pad") ? double(var_pset.at("pad")) : 0;
-        prep_params.var_info_map[var_name] =
-            PreprocessParams::VarInfo(median, norm_factor, replace_inf_value, lower_bound, upper_bound, pad);
-      }
-
-      // create data storage with a fixed size vector initilized w/ 0
-      const auto &len = input_sizes_.emplace_back(prep_params.max_length * prep_params.var_names.size());
-      data_.emplace_back(len, 0);
-    }
-  } else {
-    // otherwise use the PSet in the python config file
-    const auto &prep_pset = iConfig.getParameterSet("preprocessParams");
-    input_names_ = prep_pset.getParameter<std::vector<std::string>>("input_names");
-    for (const auto &group_name : input_names_) {
-      const auto &group_pset = prep_pset.getParameterSet(group_name);
-      auto &prep_params = prep_info_map_[group_name];
-      prep_params.var_names = group_pset.getParameter<std::vector<std::string>>("var_names");
-      prep_params.min_length = group_pset.getParameter<unsigned>("var_length");
-      prep_params.max_length = prep_params.min_length;
-      const auto &var_info_pset = group_pset.getParameterSet("var_infos");
-      for (const auto &var_name : prep_params.var_names) {
-        const auto &var_pset = var_info_pset.getParameterSet(var_name);
-        double median = var_pset.getParameter<double>("median");
-        double norm_factor = var_pset.getParameter<double>("norm_factor");
-        double replace_inf_value = var_pset.getParameter<double>("replace_inf_value");
-        double lower_bound = var_pset.getParameter<double>("lower_bound");
-        double upper_bound = var_pset.getParameter<double>("upper_bound");
-        prep_params.var_info_map[var_name] =
-            PreprocessParams::VarInfo(median, norm_factor, replace_inf_value, lower_bound, upper_bound, 0);
-      }
-
-      // create data storage with a fixed size vector initiliazed w/ 0
-      const auto &len = input_sizes_.emplace_back(prep_params.max_length * prep_params.var_names.size());
-      data_.emplace_back(len, 0);
-    }
-  }
+      debug_(iConfig.getUntrackedParameter<bool>("debugMode", false)),
+      helper_(iConfig, true, input_names_, prep_info_map_, input_shapes_) {
+  helper_.constructParticleNet();
+  data_ = helper_.data_;
```

Comment:
could pass a pointer to `data_` when initializing `helper_`, and only use it if it's non-null

Reply:
Yep good point, implemented now